### DISCUSSION TITLE
4.2.100 iOS版本bug

### DISCUSSION THREAD

**Discussion by TJCZJH** [Tencent/libpag#2128] [General] [answered]
同一个PAGView更新其中的composition，调用addLayer方法，然后调用PAGView的setComposition方法，动画经常出现多播放一次

**Comment by TJCZJH** [comment]
iOS pag 4.2.100版本，同一个pagView实例通过addLayer添加pagFile更新其中composition，然后调用PAGView的setComposition方法进行播放，动画经常出现多播放一次，回调函数onAnimationEnd会在重复播放后才调用，这个是不是4.2.100已知的bug呢

**Comment by TJCZJH** [comment]
跟repeatCount没关系，前几次播放正常，偶现一次重复播放两次

**Comment by TJCZJH** [comment]
<img width="308" alt="image" src="https://github.com/Tencent/libpag/assets/28642387/674f8a4f-ac21-4c8e-9fdc-9401c2adfa11">
该执行end回调的时候,update回调又开始执行了

**Comment by kevingpqi123** [comment, marked as answer]
建议升级至 PAG 4.3 的最新版本测试下
